moc_openstack_api_reporting.py

```diff
@@ -727,17 +727,19 @@ def main():
 
     vm_keys = list(cluster_state["vm_timestamps"].keys())
     for key in vm_keys:
-        if cluster_state["vm_timestamps"][key]["updated"] == 0:
-            del cluster_state["vm_timestamps"][key]
-        else:
-            del cluster_state["vm_timestamps"][key]["updated"]
+        if cluster_state["vm_timestamps"][key].get("updated") is not None: 
+            if cluster_state["vm_timestamps"][key]["updated"] == 0:
+                del cluster_state["vm_timestamps"][key]
+            else:
+                del cluster_state["vm_timestamps"][key]["updated"]
 
     vol_keys = list(cluster_state["vol_timestamps"].keys())
     for key in vol_keys:
-        if cluster_state["vol_timestamps"][key].get("updated") == 0:
-            del cluster_state["vol_timestamps"][key]
-        else:
-            del cluster_state["vol_timestamps"][key]["updated"]
+        if cluster_state["vol_timestamps"][key].get("updated") is not None: 
```

Comment:
You can check if a key is present in a dictionary by doing `<key> in <dict>` for example. 

```
>>> a = {'a': 1, 'b': 2}
>>> 'c' in a
False
>>> 'a' in a
True
```

So this line can be rewritten for readability as `if "updated" in cluster_state["vol_timestamps"][key]`.